src/service/video-manager-impl.js

```diff
@@ -284,43 +299,111 @@ class VideoEntry {
+    width: '0px',
+    height: '0px',
+    opacity: '0',
+  });
+
+  // The following video file is an h.264 encoded two-second video with four
+  // black frames generated with wide decoding compatibility in mind.
+  // File is 909 bytes (1212 in base64).
+  // Video file is generated by the following `ffmpeg` command:
+  //  $ffmpeg -f lavfi -i color=color=black:rate=2:size=100x100 -t 2
+  //  -profile:v baseline -preset slow
+  //  -pix_fmt yuv420p -vcodec libx264 /tmp/small.mp4
+  /*eslint-disable */
+  detectionElement.src = 'data:video/mp4;base64,AAAAIGZ0eXBpc29tAAACAGlzb21pc28yYXZjMW1wNDEAAAAIZnJlZQAAAFttZGF0AAAAMmWIhD///8PAnFAAFPf3333331111111111111111111111111111111111111111114AAAABUGaOeDKAAAABkGaVHgygAAAAAZBmnZ4MoAAAAMKbW9vdgAAAGxtdmhkAAAAAAAAAAAAAAAAAAAD6AAAB9AAAQAAAQAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAjt0cmFrAAAAXHRraGQAAAAPAAAAAAAAAAAAAAABAAAAAAAAB9AAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAABAAAAAAGQAAABkAAAAAAAkZWR0cwAAABxlbHN0AAAAAAAAAAEAAAfQAAAAAAABAAAAAAGzbWRpYQAAACBtZGhkAAAAAAAAAAAAAAAAAAAAAgAAAARVxAAAAAAALWhkbHIAAAAAAAAAAHZpZGUAAAAAAAAAAAAAAABWaWRlb0hhbmRsZXIAAAABXm1pbmYAAAAUdm1oZAAAAAEAAAAAAAAAAAAAACRkaW5mAAAAHGRyZWYAAAAAAAAAAQAAAAx1cmwgAAAAAQAAAR5zdGJsAAAAlnN0c2QAAAAAAAAAAQAAAIZhdmMxAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAGQAZABIAAAASAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGP//AAAAMGF2Y0MBQsAK/+EAGGdCwArZhz+efARAAAADAEAAAAMBA8SJmgEABWjJYPLIAAAAGHN0dHMAAAAAAAAAAQAAAAQAAAABAAAAFHN0c3MAAAAAAAAAAQAAAAEAAAAcc3RzYwAAAAAAAAABAAAAAQAAAAQAAAABAAAAJHN0c3oAAAAAAAAAAAAAAAQAAAA2AAAACQAAAAoAAAAKAAAAFHN0Y28AAAAAAAAAAQAAADAAAABbdWR0YQAAAFNtZXRhAAAAAAAAACFoZGxyAAAAAAAAAABtZGlyYXBwbAAAAAAAAAAAAAAAACZpbHN0AAAAHql0b28AAAAWZGF0YQAAAAEAAAAAR29vZ2xl';
+  /*eslint-enable */
+
+  const loadedPromise = ampdoc.whenBodyAvailable().then(body => {
+    body.appendChild(detectionElement);
+    return listenOncePromise(detectionElement, 'loadstart');
+  });
+
+  const playingPromise = loadedPromise.then(() => {
+    const playResult = detectionElement.play();
+
+    // New browsers return a promise for play call
+    if (playResult && playResult.then) {
+      return playResult;
+    } else {
+      // No play promise, wait for the `playing` event
+      return listenOncePromise(detectionElement, 'playing');
+    }
+  });
+
+  const TIMEOUT = 1000; // Allow enough time for decoding on busy/low-end CPUs
+  const timeoutPromise = timer.timeoutPromise(TIMEOUT, playingPromise);
+
+  // Unles playing promise is rejected or times out, autoplay is supported
+  const resultPromise = timeoutPromise.then(() => true, () => false);
+
+  // Remove the detection element when we know the result
+  resultPromise.then(() => {
+    detectionElement.remove();
+  });
+
+  return supportsAutoplayCache_ = resultPromise;
+}
+
+/**
+ * Clears the cache used by supportsAutoplay method.
+ *
+ * @private visible for testing.
+ */
+export function clearSupportsAutoplayCache() {
```

Comment:
Not sure if this can be tree shaken automatically but if you're only using this in tests I believe you can suffix it with `ForTesting` and it'll be dced. 

@erwinmombay @cramforce would probably know.